weldx/util.py

```diff
@@ -919,41 +919,43 @@ def xr_interp_orientation_in_time(
         Interpolated data
 
     """
-    if "time" not in dsx.coords:
-        return dsx
+    if "time" not in da.dims:
+        return da
+    if len(da.time) == 1:  # remove "time dimension" for static case
+        return da.isel({"time": 0})
 
-    times = Time(times).as_pandas_index()
-    times_ds = Time(dsx).as_pandas_index()
-    time_ref = dsx.weldx.time_ref
-
-    if len(times_ds) > 1:
-        # extract intersecting times and add time range boundaries of the data set
-        times_ds_limits = pd.Index([times_ds.min(), times_ds.max()])
-        times_union = times.union(times_ds_limits)
-        times_intersect = times_union[
-            (times_union >= times_ds_limits[0]) & (times_union <= times_ds_limits[1])
-        ]
-
-        # interpolate rotations in the intersecting time range
-        rotations_key = Rot.from_matrix(dsx.transpose(..., "c", "v").data)
-        times_key = times_ds.view(np.int64)
-        rotations_interp = Slerp(times_key, rotations_key)(
-            times_intersect.view(np.int64)
-        )
-        dsx_out = xr_3d_matrix(rotations_interp.as_matrix(), times_intersect)
-    else:
-        # TODO: this case is not really well defined, maybe avoid?
-        dsx_out = dsx
+    time = Time(time).as_pandas_index()
+    time_da = Time(da).as_pandas_index()
+    time_ref = da.weldx.time_ref
+
+    if not len(time_da) > 1:
+        raise ValueError("Invalid time format for interpolation.")
+
+    # extract intersecting times and add time range boundaries of the data set
+    times_ds_limits = pd.Index([time_da.min(), time_da.max()])
+    times_union = time.union(times_ds_limits)
+    times_intersect = times_union[
+        (times_union >= times_ds_limits[0]) & (times_union <= times_ds_limits[1])
+    ]
+
+    # interpolate rotations in the intersecting time range
+    rotations_key = Rot.from_matrix(da.transpose(..., "time", "c", "v").data)
+    times_key = time_da.view(np.int64)
+    rotations_interp = Slerp(times_key, rotations_key)(times_intersect.view(np.int64))
+    da = xr_3d_matrix(rotations_interp.as_matrix(), times_intersect)
 
     # use interp_like to select original time values and correctly fill time dimension
-    dsx_out = xr_interp_like(dsx_out, {"time": times}, fillna=True)
+    da = xr_interp_like(da, {"time": time}, fillna=True)
 
     # resync and reset to correct format
     if time_ref:
-        dsx_out.weldx.time_ref = time_ref
-    dsx_out = dsx_out.weldx.time_ref_restore()
+        da.weldx.time_ref = time_ref
+    da = da.weldx.time_ref_restore().transpose(..., "time", "c", "v")
 
-    return dsx_out.transpose(..., "c", "v")
+    if len(da.time) == 1:  # remove "time dimension" for static case
+        return da.isel({"time": 0})
```

Comment:
This is the replacement we should use instead of `return coordinates[0].values` or even `return coordinates[0]`, explicitly naming the dimension

Reply:
I agreed with you in the case of this generalized function. However, as the internal structure in the LCS was well defined, accessing per index was safe and probably faster (no lookups).